client/my-sites/domains/domain-management/settings/cards/name-servers-card.tsx

```diff
@@ -85,11 +85,23 @@ const NameServersCard = ( {
 			return false;
 		}
 
-		return nameservers.every( ( nameserver ) => {
+		return nameservers.some( ( nameserver ) => {
 			return ! nameserver || CLOUDFLARE_NAMESERVERS_REGEX.test( nameserver );
 		} );
 	};
 
+	const onlyWpcomNameservers = () => {
+		if ( ! nameservers || nameservers.length === 0 ) {
+			return false;
+		}
+
+		return (
+			nameservers.every( ( nameserver ) => {
+				return WPCOM_DEFAULT_NAMESERVERS_REGEX.test( nameserver );
+			} ) && nameservers.length === WPCOM_DEFAULT_NAMESERVERS.length
```

Comment:
Nit: The second check costs less than the first one. Would switching them work the same way?

Reply:
Technically correct, which is the best kind of correct. Certainly fixing it! Thanks :)